File: app/src/main/java/com/googlenews/googlenews/interactore/NewsFetchInteractore.java
package com.googlenews.googlenews.interactore;

import android.content.Context;
import android.util.Log;

import com.googlenews.googlenews.presenter.PresenterInterface;
import com.loopj.android.http.AsyncHttpClient;
import com.loopj.android.http.AsyncHttpResponseHandler;
import com.loopj.android.http.RequestParams;

import java.io.IOException;

import cz.msebera.android.httpclient.Header;


public class NewsFetchInteractore  {
    Context mContext;
    PresenterInterface mPresenterInterface;
    public NewsFetchInteractore(PresenterInterface presenterInterface, Context context) {
        this.mPresenterInterface=presenterInterface;
        this.mContext=context;
    }

    public  void getNewsFormGoogle(String url ,RequestParams requestParams){

        mPresenterInterface.startProgressDialog();
        AsyncHttpClient asyncHttpClient = new AsyncHttpClient();
        asyncHttpClient.get(url, requestParams, new AsyncHttpResponseHandler() {
            @Override
            public void onSuccess(int statusCode, Header[] headers, byte[] responseBody) {
                try {
                    mPresenterInterface.newsResponce(responseBody);
                    mPresenterInterface.closeProgressDialog();
                }  catch (IOException e) {
                    e.printStackTrace();
                }
                Log.i("hrController", "onSuccess: "+responseBody);
            }

            @Override
            public void onFailure(int statusCode, Header[] headers, byte[] responseBody, Throwable error) {
                Log.i("Data Available", "Failure: " +statusCode);

            }
        });

    }
}

File: app/src/main/java/com/googlenews/googlenews/interactore/NewsFetchInteractore.kt
package com.googlenews.googlenews.interactore

import android.content.Context
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.Volley
import com.googlenews.googlenews.presenter.NewsFetchPresenter
import org.json.JSONObject


class NewsFetchInteractore(newsFetchPresenter: NewsFetchPresenter, mContext: Context) {
    var mcontext: Context=mContext;
    var newsFetchPresenter=newsFetchPresenter
    fun getNewsFormGoogle(url: String) {
        val que=Volley.newRequestQueue(mcontext.applicationContext)
        val req=JsonObjectRequest(Request.Method.GET,url,null,
                Response.Listener {
                            response ->
                    callResponse(response)
        },Response.ErrorListener {
        })
        que.add(req)
    }
    private fun callResponse(response: JSONObject?) {
        newsFetchPresenter.setResponse(response);

    }
}

